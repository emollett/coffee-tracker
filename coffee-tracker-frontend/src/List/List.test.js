import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import testData from '../testdata.json';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<List data={testData}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
    const tree = renderer
      .create(<List data={testData}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('sorts the user added most recently to the top', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List data={testData}/>, div);
    expect(div.querySelector("td").textContent).toBe("6")
  });