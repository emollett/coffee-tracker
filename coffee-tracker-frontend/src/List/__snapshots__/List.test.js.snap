// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bigPadding"
>
  <div
    className="smallPadding"
  >
    <table
      className="table"
    >
      <thead>
        <tr>
          <th>
            ID
          </th>
          <th>
            Name
          </th>
          <th>
            Date Purchased
          </th>
          <th>
            Date Opened
          </th>
          <th>
            Coffee
          </th>
          <th>
            Note
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            Alice
          </td>
          <td>
            <time
              dateTime={"2018-05-13T07:01:41.522Z"}
            >
              13/05/2018
            </time>
          </td>
          <td>
            <time
              dateTime={"2018-05-13T07:01:41.522Z"}
            >
              13/05/2018
            </time>
          </td>
          <td>
            Tesco
          </td>
          <td>
            Hello
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td>
            Alice
          </td>
          <td>
            <time
              dateTime={"2018-07-13T07:01:41.522Z"}
            >
              13/07/2018
            </time>
          </td>
          <td>
            <time
              dateTime={"2018-07-13T07:01:41.522Z"}
            >
              13/07/2018
            </time>
          </td>
          <td>
            Aldi
          </td>
          <td>
            Hello
          </td>
        </tr>
        <tr>
          <td>
            3
          </td>
          <td>
            Betty
          </td>
          <td>
            <time
              dateTime={"2018-07-13T07:01:41.522Z"}
            >
              13/07/2018
            </time>
          </td>
          <td>
            <time
              dateTime={"2018-07-13T07:01:41.522Z"}
            >
              13/07/2018
            </time>
          </td>
          <td>
            Tesco
          </td>
          <td>
            Hello
          </td>
        </tr>
        <tr>
          <td>
            4
          </td>
          <td>
            Cait
          </td>
          <td>
            <time
              dateTime={"2018-06-13T07:01:41.522Z"}
            >
              13/06/2018
            </time>
          </td>
          <td>
            <time
              dateTime={"2018-06-13T07:01:41.522Z"}
            >
              13/06/2018
            </time>
          </td>
          <td>
            Tesco
          </td>
          <td>
            Hello
          </td>
        </tr>
        <tr>
          <td>
            5
          </td>
          <td>
            Cait
          </td>
          <td>
            <time
              dateTime={"2018-06-13T07:01:41.522Z"}
            >
              13/06/2018
            </time>
          </td>
          <td>
            <time
              dateTime={"2018-06-13T07:01:41.522Z"}
            >
              13/06/2018
            </time>
          </td>
          <td>
            Tesco
          </td>
          <td>
            Hello
          </td>
        </tr>
        <tr>
          <td>
            6
          </td>
          <td>
            Alice
          </td>
          <td>
            <time
              dateTime={"2019-07-13T07:01:41.522Z"}
            >
              13/07/2019
            </time>
          </td>
          <td>
            <time
              dateTime={"2019-07-13T07:01:41.522Z"}
            >
              13/07/2019
            </time>
          </td>
          <td>
            Aldi
          </td>
          <td>
            Hello
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
